server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -565,7 +572,7 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
         else
           mmfi.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
 
-        while (mmfi.hasTop()) {
+        while (!yield.hasYielded() && mmfi.hasTop()) {
```

Comment:
I was thinking it would be cleaner if the contract was : *when an iterator sets a yield key that its hasTop() method should start returning false*

I think has the following benefits :
  * No requirement for intermediate iterator (that push the YieldCallback down) to check  yield.hasYielded().  I think makes it easier for simple iterators to support pushing the YieldCallback down.  If an iterator cares, it can still check.
  * May be more efficient to only check `hasTop()` instead of always checking `yield.hasYielded()` and `hasTop()`

When `hasTop()` returns false, then the top level loop can check `yield.hasYielded()` as it currently does.